model/src/main/java/uk/gov/hmcts/payment/api/service/PciPalPaymentService.java

```diff
@@ -74,6 +133,79 @@ else if (serviceType.equalsIgnoreCase(SERVICE_TYPE_PROBATE))
         });
     }
 
+    public TelephonyProviderAuthorisationResponse getTelephonyProviderLink(PciPalPaymentRequest pciPalPaymentRequest, TelephonyProviderAuthorisationResponse telephonyProviderAuthorisationResponse, String serviceType, String returnURL) {
+        return withIOExceptionHandling(() -> {
+
+            String flowId = getFlowId(serviceType);
+            LOG.info("flowId: {}   launchURL: {}   viewIdURL: {}   callbackUrl: {}   returnURL: {} ", flowId, launchURL, viewIdURL, callbackUrl, returnURL);
+            HttpPost httpPost = new HttpPost(launchURL);
+            httpPost.addHeader(CONTENT_TYPE, APPLICATION_JSON.toString());
+            httpPost.addHeader(authorizationHeader(telephonyProviderAuthorisationResponse.getAccessToken()));
+            TelephonyProviderLinkIdRequest telephonyProviderLinkIdRequest = TelephonyProviderLinkIdRequest.telephonyProviderLinkIdRequestWith().FlowId(flowId)
+                .InitialValues(TelephonyProviderLinkIdRequest.InitialValues.initialValuesWith()
+                    .amount(new BigDecimal(pciPalPaymentRequest.getOrderAmount()).movePointRight(2).toString())
+                    .callbackURL(callbackUrl)
+                    .returnURL(returnURL)
+                    .orderId(pciPalPaymentRequest.getOrderReference())
+                    .currencyCode("GBP")
+                    .build())
+                    .build();
+
+            Gson gson = new Gson();
+            StringEntity entity = new StringEntity(gson.toJson(telephonyProviderLinkIdRequest));
+            httpPost.setEntity(entity);
+            HttpResponse response = httpClient.execute(httpPost);
+            TelephonyProviderLinkIdResponse telephonyProviderLinkIdResponse = objectMapper.readValue(response.getEntity().getContent(), TelephonyProviderLinkIdResponse.class);
+            telephonyProviderAuthorisationResponse.setNextUrl(viewIdURL + telephonyProviderLinkIdResponse.getId()+"/framed");
+
+            return telephonyProviderAuthorisationResponse;
+        });
+    }
+
+    private String getFlowId(String serviceType) {
```

Comment:
I think using a Static HashMap is a better choice here to avoid several lines of codes

Reply:
Fixed.